src/cargo/ops/registry.rs

```diff
@@ -296,6 +296,18 @@ pub fn registry_login(config: &Config, token: String) -> CargoResult<()> {
     config::save_credentials(config, token)
 }
 
+pub fn registry_whoami(config: &Config, index: Option<String>) -> CargoResult<()> {
+    let RegistryConfig { token, .. } = registry_configuration(config)?;
+    let (mut registry, _) = registry(config, token.clone(), index)?;
+
+    match registry.whoami() {
+        Ok(user) => config.shell().status("Whoami", format!("Currently logged in as `{}`", user.login))?,
```

Comment:
I wonder if this wording is optimal. "Currently logged in as", given to a web-accustomed audience, sounds like the status of a temporary session. We could also be more clear about where this `user` is actually used.

FWIW "Crates will be published by user \`{}\`"

Reply:
@lukaslueg I get where you're coming from, and normally I'd agree. Two things are hanging me up though. The first is that users already establish local credentials by running `cargo login` (normally), so there's already some precedent there. Additionally, while `publish`/`yank` are probably the two most common commands, there's certainly more to do that requires authentication.

If you wanted to move away from `logged in`, maybe `Authenticated as {}` or `Operating as {}`?

Reply:
I'd suggest mirroring what `whoami` or `id -un` does and just print out the username, with no further formatting. This has the added benefit that the command can be used in scripts more easily, without having to do a `awk`-pass. That is

```
$ cargo whoami
leeroyjenkins
```
and
```
$ cargo whoami
[some TokenMissing-message to stderr]
$ echo $?
1
```

Reply:
I thought about that as well while implementing this. The main reason I didn't is because it doesn't match how the rest of cargo generally outputs info to STDOUT

Reply:
#4646 comes with a small change in `config.shell()` to determine if `stderr` is actually human-readable. One can extend that to `stdout` and output an unformatted string if stdout is hooked up. If wonder if this is surprising behavior, though